### PR TITLE
Move unit enums to Qgis, promote to enum class

### PR DESCRIPTION
**Pull request by nyalldawson**
FYI @3nids

### CODE REVIEW

**Review by 3nids — APPROVED** [review]

File: src/core/qgis.h [hunk 1]
```diff
@@ -2919,7 +2919,7 @@ class CORE_EXPORT Qgis
      */
     enum class AreaUnit SIP_MONKEYPATCH_SCOPEENUM_UNNEST( QgsUnitTypes, AreaUnit ) : int
       {
-      SquareMeters = 0 SIP_MONKEYPATCH_COMPAT_NAME( AreaSquareMeters ), //!< Square meters
```
Comment (3nids) [inline src/core/qgis.h]: that would have deserved a fix in sipify 😛

Comment (nyalldawson) [reply src/core/qgis.h]: I tried, and failed :laughing:

File: src/gui/layout/qgslayoutunitscombobox.h [hunk 2]
```diff
@@ -82,10 +82,18 @@ class GUI_EXPORT QgsLayoutUnitsComboBox : public QComboBox
...
+     * Emitted when the \a unit is changed.
+     */
+    void unitChanged( Qgis::LayoutUnit unit );
+#endif
+
     /**
      * Emitted when the \a unit is changed.
      */
-    void changed( Qgis::LayoutUnit unit );
+    void changed( int );
```
Comment (3nids) [inline src/gui/layout/qgslayoutunitscombobox.h:96]: add a note to drop in QGIS 4?

Comment (nyalldawson) [reply src/gui/layout/qgslayoutunitscombobox.h:96]: Well, it depends on whether sip fixes this. (It's unrelated to the qgis compatibility code)

**Pull request merged**